### PR TITLE
122 work on input sanitization

### PR DESCRIPTION
**Pull request by Chocolat2**
Implement input sanitization on email, phone number and numeric values entered by user.

### CODE REVIEW

File: app/src/main/java/com/android/PetPamper/StringSanitizer.kt [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    val sanitizedPhone = cleanedPhone.replace(Regex("""\D"""),"")
+    return  sanitizedPhone
+  }
+
+  fun sanitizeFloat(str:String):String{
+    val cleanedNumber = str.replace(Regex("""[,'.]"""),",").reversed().replaceFirst(",",".")
+    val sanitizedNumber = cleanedNumber.replace(Regex("""[^0-9,]"""),"")
+    return  sanitizedNumber
+  }
+
```
Comment (karinerafla) [inline app/src/main/java/com/android/PetPamper/StringSanitizer.kt:26]: Logic is a bit confusion (reverse, replace) but sanitisation is well done.

**Review by karinerafla — APPROVED** [review]
The code is well-structured and effectively sanitises different types of strings. Minor adjustments mentioned would've made the code perfect. However, very good implementation. Thank you.

File: app/src/main/java/com/android/PetPamper/StringSanitizer.kt [hunk 2]
```diff
@@ -0,0 +1,109 @@
+package com.android.PetPamper
+
+import java.util.regex.Pattern
```
Comment (B6Amine) [inline app/src/main/java/com/android/PetPamper/StringSanitizer.kt:3]: Combine multiple replace calls into a single regex replacement where possible.
here's a suggestion of how we can do it 
```suggestion

object StringSanitizer {
  private val emailPattern = Regex("""[^a-zA-Z0-9.!#$%&'*+/=?^_`{|}~@-]""")
  private val redundantDotPattern = Regex("""\.{2,}""")
  private val phonePattern = Regex("""[^0-9]""")
  private val floatPattern = Regex("""[^0-9,.]""")
  private val intPattern = Regex("""[^0-9]""")

  fun sanitizeEmail(str: String): String {
    return str.replace(emailPattern, "").replace(redundantDotPattern, ".")
  }

  fun sanitizePhone(str: String): String {
    return if (str.startsWith("+")) str.replaceFirst("+", "00").replace(phonePattern, "") else str.replace(phonePattern, "")
  }

  fun sanitizeFloat(str: String): String {
    return str.replace(Regex("""[,'.]"""), "").replace(',', '.')
  }

  fun sanitizeInt(str: String): String {
    return str.replace(intPattern, "")
  }
}

```

**Review by B6Amine — APPROVED** [review]
The way u did the sanitization looks well to me but ther's an optimization that we can do

**Pull request merged**